File: uni/data-structure/T01/sorting_specific_array.py
limit = 200000


def getNumbers():
    list(map(int, input().split()))


(mn, mx) = getNumbers()
numbers = getNumbers()

nTable = [0 for _ in range(0, 200)]

for n in numbers:
    nTable[n-mn] += 1

c = 0
for i, r in enumerate(nTable):
    for _ in range(0, r):
        if c < limit:
            print(mn+i, end=" ")
        else:
            quit()

        c += 1

File: uni/data-structure/T01/sorting_specific_array.nim
import std/[sequtils, strutils, strscans]

type RepeatNumbers = array[200, int]
const limit = 200000

when isMainModule:
  let
    (_, mn, mx) = stdin.readline.scanTuple("$i $i")
    numbers = stdin.readLine.splitWhitespace.map(parseInt)

  var nTable: RepeatNumbers

  for n in numbers:
    inc nTable[n-mn]

  var c = 0
  for i, r in nTable:
    for _ in 1..r:
      if c < limit:
        stdout.write mn+i, " "
      else:
        quit()

      inc c
